common/sockets/socket.hpp

```diff
@@ -112,7 +112,7 @@ class Socket : public BasicSocket
 public:
 
 	using BasicSocket::BasicSocket;
-	Socket(const Socket& other){}
+	Socket(const Socket& other) : BasicSocket(other) {}
```

Comment:
This is not semantically equivalent?

Reply:
Before it called the default constructor for `BasicSocket`, but was that really the intended purpose? The warning we got here was `base class 'class BasicSocket' should be explicitly initialized in the copy constructor`.